src/modules/Dashboard/components/ActivityAssignDrawer/AssignmentsTable/AssignmentsTable.tsx

```diff
@@ -0,0 +1,217 @@
+import { useTranslation } from 'react-i18next';
+import { useCallback, useMemo } from 'react';
+import { Box, Button } from '@mui/material';
+
+import {
+  DashboardTableProps,
+  FullTeamSearchType,
+  ParticipantDropdownOption,
+} from 'modules/Dashboard/components';
+import { useFeatureFlags } from 'shared/hooks/useFeatureFlags';
+import { Svg } from 'shared/components';
+import { variables } from 'shared/styles';
+
+import { AssignmentsTableProps } from './AssignmentsTable.types';
+import { StyledTable, StyledTableContainer } from './AssignmentsTable.styles';
+import { AssignmentDropdown } from './AssignmentDropdown';
+import { ActivityAssignment } from '../ActivityAssignDrawer.types';
+import { getHeadCells } from './AssignmentsTable.utils';
+
+export const AssignmentsTable = ({
+  allParticipants,
+  participantsOnly,
+  fullAccountParticipantsAndTeamMembers,
+  teamMembersOnly,
+  handleSearch,
+  assignments,
+  onChange,
+  isReadOnly,
+  errors,
+  'data-testid': dataTestId,
+}: AssignmentsTableProps) => {
+  const { t } = useTranslation('app', { keyPrefix: 'activityAssign' });
+  const {
+    featureFlags: { enableParticipantMultiInformant },
+  } = useFeatureFlags();
+  const lastAssignment = assignments[assignments.length - 1];
+  const showAddButton =
+    !isReadOnly && lastAssignment?.respondentSubjectId && lastAssignment?.targetSubjectId;
+
+  const handleChange = useCallback(
+    ({ respondentSubjectId, targetSubjectId }: ActivityAssignment, index: number) => {
+      const updatedAssignments = [...assignments];
+
+      if (respondentSubjectId !== undefined || targetSubjectId !== undefined) {
+        if (respondentSubjectId !== undefined) {
+          // Set as self-assigned if selected respondent is a full account and either:
+          // - no target subject has been selected yet, or
+          // - previous assignment was a self-assignment
+          const respondent = allParticipants.find(({ id }) => id === respondentSubjectId);
+          if (respondent?.userId && !respondent.isTeamMember) {
+            if (
+              !assignments[index].targetSubjectId ||
+              assignments[index].targetSubjectId === assignments[index].respondentSubjectId
+            ) {
+              updatedAssignments[index].targetSubjectId = respondentSubjectId;
+            }
+          }
```

Comment:
This doesn't affect functionality and is more of a note on your comment, it's also not in any of the AC so feel free to ignore.

From the comment, it sounds like `Self` should only be auto selected if the previous assignment was `Self`, but there's no logic comparing to previous index and adding a new assignment always results in `Self` (**except** when you select a target first). 
https://github.com/user-attachments/assets/8ab9bde8-8aee-4b70-9bf6-b7a8e760f40b

Reply:
Oh, in that comment, "previous assignment" is referring to the previously selected (or _currently_ selected) value in the dropdown of _that same row_ – not the assignment that is shown in the _previous row index_ of the table. I see how that's ambiguous. I'll update the comment to clarify.

Correct that this behaviour wasn't explicitly part of the AC, but it felt like a natural choice to me for "Self" to remain selected in the right column when the respondent changes in the left column, even though persisting "Self" in the subject column technically means you're changing the target subject ID in this instance.